### PR TITLE
Update isDomainActive condition to honor pending active statue

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Update isDomainActive condition to honor pending active statue

**Why?**
Support graceful failover

**How did you test it?**
unit test

**Potential risks**

### CODE REVIEW

File: common/cache/domainCache.go [hunk 1]
```diff
@@ -675,6 +677,7 @@ func (c *domainCache) buildEntryFromRecord(
 	newEntry.failoverVersion = record.FailoverVersion
 	newEntry.isGlobalDomain = record.IsGlobalDomain
 	newEntry.failoverNotificationVersion = record.FailoverNotificationVersion
+	newEntry.failoverEndTime = record.FailoverEndTime
```
Comment (vancexu) [inline common/cache/domainCache.go:680]: record.F or .f?

File: common/cache/domainCache.go [hunk 2]
```diff
@@ -131,6 +131,7 @@ type (
 		failoverVersion             int64
 		isGlobalDomain              bool
 		failoverNotificationVersion int64
+		failoverEndTime             *int64
```
Comment (vancexu) [inline common/cache/domainCache.go:134]: do we need 3 states for failoverEndTime? or make it int64 is enough?

File: common/cache/domainCache.go (hunk 1 above)
Comment (yux0) [reply common/cache/domainCache.go:680]: This GetDomainResponse is defined in a different package. Have to use F

File: common/cache/domainCache.go (hunk 2 above)
Comment (yux0) [reply common/cache/domainCache.go:134]: Yes. I don't want to use a special value to indicate the end time is not set. Just use a pointer here.

**Pull request merged**